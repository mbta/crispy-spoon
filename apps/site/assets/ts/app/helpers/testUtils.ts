/* eslint-disable */
import toJson, { Json } from "enzyme-to-json";
import { ReactWrapper } from "enzyme";
import {
  IndexedRouteStop,
  StopTree
} from "../../schedule/components/__schedule";

export const testRouteStopListFromStopTree = (
  tree: StopTree
): IndexedRouteStop[] =>
  Object.values(tree.byId).map((node, index) => ({
    ...node.value,
    routeIndex: index
  }));

export const createReactRoot = (): void => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
};

export const enzymeToJsonWithoutProps = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false,
    map: info => {
      if (!info.props) {
        return info;
      }

      // @ts-ignore is a <style></style> declaration
      if (info.node.type === "style") {
        return info;
      }
      // @ts-ignore is an element in a <noscript></noscript> declaration
      if (info.node.nodeType == undefined) {
        // @ts-ignore
        return info;
      }
      //@ts-ignore is a React component (not HTML element)
      if (info.node.nodeType.toString().includes("function")) {
        // @ts-ignore
        info.props = undefined;
      }

      return info;
    }
  });

export const toJsonEnzyme = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false
  });
