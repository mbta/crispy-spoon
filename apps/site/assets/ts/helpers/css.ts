import {
  isABusRoute,
  isACommuterRailRoute,
  isAGreenLineRoute,
  isASilverLineRoute
} from "../models/route";
import { Route } from "../__v3api";

export const routeToCSSClass = (route: string): string =>
  route.toLowerCase().replace(/[ _]/g, "-");

export const busClass = (route: Route): string =>
  isABusRoute(route) && !isASilverLineRoute(route) ? "bus-route-sign" : "";

export const routeToModeName = (route: Route): string => {
  if (isACommuterRailRoute(route)) return "commuter-rail";
  if (route.type === 4) return "ferry";
  if (route.id === "Red" || route.id === "Mattapan") return "red-line";
  if (route.id === "Orange") return "orange-line";
  if (route.id === "Blue") return "blue-line";
  if (isAGreenLineRoute(route)) return "green-line";
  if (isASilverLineRoute(route)) return "silver-line";
  if (isABusRoute(route)) return "bus";
  return "unknown";
};

export const routeBgClass = (route: Route): string =>
  `u-bg--${routeToModeName(route)}`;
