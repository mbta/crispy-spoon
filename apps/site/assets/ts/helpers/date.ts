// Returns a date in the form of "August 7, 2019"
export const formattedDate = (unformatted: string): string => {
  const [year, month, day] = unformatted
    .split(/-/)
    .map(part => Number.parseInt(part, 10));

  // Remember that months in JS are 0-indexed for some reason, hence "month - 1".
  const parsedDate = new Date(year, month - 1, day);
  return parsedDate.toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC"
  });
};

// Returns a date in the form of "Aug 7"
export const shortDate = (date: Date): string =>
  date.toLocaleDateString("en-US", {
    month: "short",
    day: "numeric",
    timeZone: "UTC"
  });

export default formattedDate;
