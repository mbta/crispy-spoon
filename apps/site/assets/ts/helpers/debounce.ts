// Debounce is meant to be used with event listeners. Use it to delay a call
// until after a burst of events, such as keyboard input.
//
//   usage:
//
//     const callback = (ev: Event) => {}
//     document.addEventListener("keyup", debounce(callback, 250))
//

const debounce = (callback: Function, delay: number): EventListener => {
  let timer: number | undefined;

  return (...args) => {
    window.clearTimeout(timer);
    timer = window.setTimeout(() => callback(...args), delay);
  };
};

export const debouncePromise = (
  callback: Function,
  delay: number
): Function => {
  let timerId: number | undefined;

  return function debounced(...args: unknown[]) {
    if (timerId) {
      window.clearTimeout(timerId);
    }

    return new Promise(resolve => {
      timerId = window.setTimeout(() => resolve(callback(...args)), delay);
    });
  };
};

export default debounce;
