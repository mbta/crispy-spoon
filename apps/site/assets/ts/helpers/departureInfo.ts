import { concat, difference, keyBy, keys, map, sortBy } from "lodash";
import { PredictionWithTimestamp } from "../models/perdictions";
import { ScheduleWithTimestamp } from "../models/schedules";
import { isCancelled, isDelayed } from "./prediction-helpers";
import { DepartureInfo } from "../models/departureInfo";

const departureInfoToTime = (departureInfo: DepartureInfo): Date => {
  // If there isn't a prediction there should be a schedule
  return departureInfo.prediction
    ? departureInfo.prediction.time
    : departureInfo.schedule!.time;
};

const mergeIntoDepartureInfo = (
  schedules: ScheduleWithTimestamp[],
  predictions: PredictionWithTimestamp[]
): DepartureInfo[] => {
  const schedulesByTripId = keyBy(schedules, sch => sch.trip.id);
  const predictionsByTripId = keyBy(predictions, prd => prd.trip.id);

  const scheduleTripIds = keys(schedulesByTripId);
  const predictionTripIds = keys(predictionsByTripId);

  // All the schedules with and without predictions
  const scheduleDirectionInfo = map(
    schedulesByTripId,
    (schedule, tripId): DepartureInfo => {
      // prediction could be undefined
      const prediction = predictionsByTripId[tripId];
      return {
        prediction: prediction,
        schedule: schedule,
        isCancelled: isCancelled(prediction),
        isDelayed: isDelayed(prediction, schedule)
      };
    }
  );

  const unmatchedPredictionTripIds = difference(
    predictionTripIds,
    scheduleTripIds
  );
  // All the predictions with no schedules
  const predictionDirectionInfo = map(
    unmatchedPredictionTripIds,
    (tripId: string): DepartureInfo => {
      const prediction = predictionsByTripId[tripId];
      return {
        prediction: prediction,
        isCancelled: isCancelled(prediction)
      } as DepartureInfo;
    }
  );

  const departureInfos = concat(scheduleDirectionInfo, predictionDirectionInfo);
  const sortedDepartureInfo = sortBy(departureInfos, di => {
    // prioritize sorting by predictions
    if (di.prediction) {
      return di.prediction.time;
    }
    return di.schedule?.time;
  });

  return sortedDepartureInfo;
};

export { mergeIntoDepartureInfo, departureInfoToTime };
