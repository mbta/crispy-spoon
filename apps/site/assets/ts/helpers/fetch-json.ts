export const isFetchFailed = <T>(r: FetchResult<T>): r is FetchFailed =>
  r && typeof r === "object" && "statusText" in r;

export const throwIfFetchFailed = <T>(result: FetchResult<T>): T => {
  if (isFetchFailed(result)) {
    throw new Error(`Fetch failed: ${result}`);
  }

  return result;
};

export type FetchFailed = Response;
export type FetchResult<T> = T | FetchFailed;
export const fetchJson = async <T>(
  ...args: Parameters<typeof fetch>
): Promise<FetchResult<T>> => {
  const resp = await fetch(...args);
  if (!resp.ok) {
    return resp;
  }

  return resp.json();
};

export const fetchJsonOrThrow = async <T>(
  ...args: Parameters<typeof fetch>
): Promise<T> => fetchJson<T>(...args).then(throwIfFetchFailed);
