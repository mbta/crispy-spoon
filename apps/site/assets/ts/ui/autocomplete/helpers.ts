import { omit } from "lodash";
import { OnSubmitParams } from "@algolia/autocomplete-core";
import { OnStateChangeProps } from "@algolia/autocomplete-js";
import {
  ContentItem,
  Item,
  LocationItem,
  PopularItem,
  RouteItem,
  StopItem
} from "./__autocomplete";
import { isLGDown } from "../../helpers/media-breakpoints";

export function isStopItem(x: Item): x is StopItem {
  return Object.keys(x).includes("stop");
}

export function isRouteItem(x: Item): x is RouteItem {
  return Object.keys(x).includes("route");
}

export function isContentItem(x: Item): x is ContentItem {
  return Object.keys(x).includes("_content_type");
}

export const getTitleAttribute = (item: Item): string[] => {
  if (isStopItem(item)) {
    return ["stop", "name"];
  }
  if (isRouteItem(item)) {
    return ["route", "name"];
  }

  // @ts-ignore
  // eslint-disable-next-line no-underscore-dangle
  return Object.keys(item._highlightResult);
};

const navStateChange: (props: OnStateChangeProps<Item>) => void = ({
  state
}) => {
  // grey out the page and disable scrolling when search is open
  if (isLGDown()) {
    if (state.isOpen) {
      document.documentElement.dataset.navOpen = "true";
    } else {
      delete document.documentElement.dataset.navOpen;
    }
  }
};

export const STATE_CHANGE_HANDLERS: Record<
  string,
  (props: OnStateChangeProps<Item>) => void
> = {
  nav: navStateChange
};

export const SUBMIT_HANDLERS: Record<
  string,
  (props: OnSubmitParams<Item>) => void
> = {
  to_search_page({ state }) {
    window.Turbolinks.visit(`/search?query=${state.query}`);
  }
};

export type WithUrls<T> = T & { urls: Record<string, string> };

// The backend returns all possible URLs, use urlType to get the desired one
export const itemWithUrl = (
  initialItem: WithUrls<LocationItem | PopularItem>,
  urlType: string
): Omit<typeof initialItem, "urls"> => {
  const item = omit(initialItem, "urls");
  item.url = initialItem.urls[urlType];
  return item;
};
