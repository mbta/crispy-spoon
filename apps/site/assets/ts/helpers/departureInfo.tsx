import React from "react";
import {
  chain,
  concat,
  difference,
  find,
  keyBy,
  keys,
  map,
  sortBy
} from "lodash";
import { PredictionWithTimestamp } from "../models/predictions";
import { ScheduleWithTimestamp } from "../models/schedules";
import { isCancelled, isDelayed, isSkipped } from "./prediction-helpers";
import { DepartureInfo } from "../models/departureInfo";
import {
  isABusRoute,
  isACommuterRailRoute,
  isSubwayRoute
} from "../models/route";
import { DirectionId, Route } from "../__v3api";
import DisplayTime from "../stop/components/DisplayTime";
import { getInfoKey } from "../stop/models/displayTimeConfig";
import { RoutePatternWithPolyline } from "../models/route-patterns";

export const SUBWAY = "subway";
export const BUS = "bus";
export const COMMUTER_RAIL = "commuter_rail";
export const FERRY = "ferry";

const toRouteMode = (
  route: Route
): typeof SUBWAY | typeof BUS | typeof COMMUTER_RAIL | typeof FERRY => {
  if (isSubwayRoute(route)) {
    return SUBWAY;
  }
  if (isABusRoute(route)) {
    return BUS;
  }
  if (isACommuterRailRoute(route)) {
    return COMMUTER_RAIL;
  }
  return FERRY;
};

const departureInfoToTime = (departureInfo: DepartureInfo): Date => {
  // If there isn't a prediction there should be a schedule
  return departureInfo.prediction && departureInfo.prediction.time
    ? departureInfo.prediction.time
    : departureInfo.schedule!.time;
};

const getNextUnCancelledDeparture = (
  departureInfos: DepartureInfo[]
): DepartureInfo | undefined => {
  return find(departureInfos, (di: DepartureInfo) => !di.isCancelled);
};

const displayInfoContainsPrediction = (
  departureInfo: DepartureInfo
): boolean => {
  return !!departureInfo.prediction;
};

const mergeIntoDepartureInfo = (
  schedules: ScheduleWithTimestamp[],
  predictions: PredictionWithTimestamp[]
): DepartureInfo[] => {
  const schedulesByTripId = keyBy(schedules, "trip.id");
  const predictionsByTripId = keyBy(predictions, "trip.id");
  const scheduleTripIds = keys(schedulesByTripId);
  const predictionTripIds = keys(predictionsByTripId);

  // All the schedules with and without predictions
  const scheduleDirectionInfo = map(
    schedulesByTripId,
    (schedule, tripId): DepartureInfo => {
      // prediction could be undefined
      const prediction = predictionsByTripId[tripId];
      return {
        prediction,
        schedule,
        route: schedule.route,
        trip: schedule.trip,
        isCancelled: isCancelled(prediction),
        isSkipped: isSkipped(prediction),
        isDelayed: isDelayed(prediction, schedule),
        routeMode: toRouteMode(schedule.route)
      };
    }
  );

  const unmatchedPredictionTripIds = difference(
    predictionTripIds,
    scheduleTripIds
  );
  // All the predictions with no schedules
  const predictionDirectionInfo = map(
    unmatchedPredictionTripIds,
    (tripId: string): DepartureInfo => {
      const prediction = predictionsByTripId[tripId];
      return {
        prediction,
        route: prediction.route,
        trip: prediction.trip,
        isCancelled: isCancelled(prediction),
        isSkipped: isSkipped(prediction),
        routeMode: toRouteMode(prediction.route)
      } as DepartureInfo;
    }
  );

  const departureInfos = concat(scheduleDirectionInfo, predictionDirectionInfo);
  const sortedDepartureInfo = sortBy(departureInfos, di => {
    // prioritize sorting by predictions
    if (di.prediction && di.prediction.time) {
      return di.prediction.time;
    }
    return di.schedule?.time;
  });

  return sortedDepartureInfo;
};

const isAtDestination = (
  stopName: string,
  route: Route,
  directionId: DirectionId
): boolean => {
  const destination = route.direction_destinations[directionId];
  if (destination) {
    return destination.includes(stopName) && !destination.includes(" or ");
  }
  return false;
};

const isCommuterRail = (departureInfo: DepartureInfo): boolean => {
  return departureInfo?.routeMode === "commuter_rail";
};

const DefaultWrapper: React.FunctionComponent<unknown> = ({ children }) => (
  <li>{children}</li>
);
/**
 * From a list of `DepartureInfo[]`, generate a list of `<DisplayTime />`.
 * Each `<DisplayTime />` will recieve the `isCR` and `targetDate` props.
 *
 * This function can optionally
 * - wrap each `<DisplayTime />` in a specified wrapper element, `WrapperEl`
 * - truncate the list to a desired `listLength` value
 */
const departuresListFromInfos = (
  departureInfos: DepartureInfo[],
  isCR: boolean,
  targetDate?: Date,
  listLength?: number,
  omitCancelledAndSkipped = false,
  WrapperEl: typeof DefaultWrapper = DefaultWrapper
): React.ReactElement[] => {
  // optional cutoff time, before which we won't show schedules.
  // just used with subway for now.
  const predictionTimeCutoff = chain(departureInfos)
    .filter(d => d.routeMode === SUBWAY)
    .maxBy("prediction.time")
    .value()?.prediction!.time;

  return chain(departureInfos)
    .reject(
      departure =>
        omitCancelledAndSkipped &&
        (!!departure.isCancelled || !!departure.isSkipped)
    )
    .omitBy(
      ({ prediction, schedule }) =>
        // omit schedule-only departures that are before latest prediction time
        predictionTimeCutoff &&
        !prediction &&
        schedule &&
        schedule.time <= predictionTimeCutoff
    )
    .map(d => (
      <WrapperEl key={getInfoKey(d)}>
        <DisplayTime departure={d} isCR={isCR} targetDate={targetDate} />
      </WrapperEl>
    ))
    .slice(0, listLength)
    .value();
};

const departureInfoInRoutePatterns = (
  departureInfo: DepartureInfo,
  routePatterns: RoutePatternWithPolyline[]
): boolean => {
  const isMatch = !!routePatterns.find(
    rp => rp.id === departureInfo.trip.route_pattern_id
  );
  // This second case is needed for the edge case where the route pattern is
  // canonical but the schedules/predictions are not serving canonical route
  // patterns (e.g. due to shuttles or other disruptions)
  const isSimilar = !!routePatterns.find(
    rp =>
      rp.headsign === departureInfo.trip.headsign &&
      rp.route_id === departureInfo.route.id
  );
  return isMatch || isSimilar;
};

export {
  mergeIntoDepartureInfo,
  departureInfoToTime,
  displayInfoContainsPrediction,
  getNextUnCancelledDeparture,
  isAtDestination,
  isCommuterRail,
  departuresListFromInfos,
  departureInfoInRoutePatterns
};
