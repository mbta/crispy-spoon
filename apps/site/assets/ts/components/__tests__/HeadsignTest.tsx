import React from "react";
import renderer from "react-test-renderer";
import HeadsignComponent from "../Headsign";
import { createReactRoot } from "../../app/helpers/testUtils";
import { Headsign } from "../../__v3api";
import { shallow } from "enzyme";

/* eslint-disable camelcase */

it("it renders 2 predictions", () => {
  const headsign: Headsign = {
    name: "Watertown",
    headsign: "Watertown",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["5:00", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={1} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders 1 prediction for CR", () => {
  const headsign: Headsign = {
    name: "Watertown",
    headsign: "Watertown",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["5:00", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  const wrapper = shallow(
    <HeadsignComponent headsign={headsign} condensed={false} routeType={2} />
  );

  expect(wrapper.find(".m-tnm-sidebar__schedule")).toHaveLength(1);
});

it("it renders scheduled time when prediction is null", () => {
  const headsign: Headsign = {
    name: "Watertown",
    headsign: "Watertown",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: null
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={1} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it splits the headsign name when it contains 'via' ", () => {
  const headsign: Headsign = {
    name: "Watertown via Harvard Square",
    headsign: "Watertown via Harvard Square",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["5:00", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={1} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders a status and train name for Commuter Rail", () => {
  const headsign: Headsign = {
    name: "Framingham",
    headsign: "Framingham",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["4:30", " ", "PM"],
          status: "On time",
          track: null
        }
      }
    ],
    train_number: "591"
  };

  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={2} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders a status and train name for Commuter Rail with track number if available", () => {
  const headsign: Headsign = {
    name: "Framingham",
    headsign: "Framingham",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["4:30", " ", "PM"],
          status: "Now boarding",
          track: "1"
        }
      }
    ],
    train_number: "591"
  };

  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={2} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders uncondensed bus headsign name as --small", () => {
  const headsign: Headsign = {
    name: "Watertown via Copley (Express)",
    headsign: "Watertown via Copley (Express)",
    times: [
      {
        delay: 0,
        scheduled_time: ["7:00", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["7:10", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={3} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it displays delayed status for CR", () => {
  const headsign: Headsign = {
    name: "Delayed Train",
    headsign: "Delayed Train",
    train_number: "404",
    times: [
      {
        delay: 5,
        scheduled_time: ["7:00", " ", "PM"],
        prediction: {
          schedule_relationship: null,
          time: ["7:03", " ", "PM"],
          status: null,
          track: null
        }
      }
    ]
  };
  createReactRoot();
  const tree = renderer
    .create(
      <HeadsignComponent headsign={headsign} condensed={false} routeType={2} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it displays cancelled status for CR", () => {
  const headsign: Headsign = {
    name: "Cancelled Train",
    headsign: "Cancelled Train",
    train_number: "404",
    times: [
      {
        delay: 5,
        scheduled_time: ["7:00", " ", "PM"],
        prediction: {
          schedule_relationship: "cancelled",
          time: ["7:03", " ", "PM"],
          status: null,
          track: null
        }
      }
    ]
  };
  const wrapper = shallow(
    <HeadsignComponent headsign={headsign} condensed={false} routeType={2} />
  );

  expect(
    wrapper.find(".m-tnm-sidebar__time-number.strikethrough").exists()
  ).toBeTruthy();
});
