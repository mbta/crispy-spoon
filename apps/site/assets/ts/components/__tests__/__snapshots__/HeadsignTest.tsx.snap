// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays delayed status for CR 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Delayed Train
    </div>
    <div
      className="m-tnm-sidebar__headsign-train"
    >
      Train 404
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail  "
      >
        <div
          className=" m-tnm-sidebar__time-number--delayed"
        >
          7:00 PM
        </div>
        <div
          className=" m-tnm-sidebar__time-number"
        >
          7:03 PM
        </div>
        <div
          className="m-tnm-sidebar__status"
        >
          Delayed 5 min
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it displays on time status with the scheduled time and not the prediction time for CR 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Worcester
    </div>
    <div
      className="m-tnm-sidebar__headsign-train"
    >
      Train 511
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail  "
      >
        <div
          className=" m-tnm-sidebar__time-number"
        >
          10:21 AM
        </div>
        <div
          className="m-tnm-sidebar__status"
        >
          On time
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders 2 predictions 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Watertown
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a status and train name for Commuter Rail 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Framingham
    </div>
    <div
      className="m-tnm-sidebar__headsign-train"
    >
      Train 591
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail  "
      >
        <div
          className=" m-tnm-sidebar__time-number"
        >
          4:30 PM
        </div>
        <div
          className="m-tnm-sidebar__status"
        >
          On time
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a status and train name for Commuter Rail with track number if available 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Framingham
    </div>
    <div
      className="m-tnm-sidebar__headsign-train"
    >
      Train 591
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail  "
      >
        <div
          className=" m-tnm-sidebar__time-number"
        >
          4:30 PM
        </div>
        <div
          className="m-tnm-sidebar__status"
        >
          Now boarding track 1
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders in chronological order when first time is later than second time 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Watertown
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders in chronological order when second equals arriving 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Watertown
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          arriving
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          5:00
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          PM
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders in chronological order when times are hh:mm format 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Watertown
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          5:00
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          PM
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          4:30
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          PM
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders scheduled time when prediction is null 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Watertown
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--schedule"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          4:30
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          PM
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders uncondensed bus headsign name as --small 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--small notranslate"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via notranslate"
    >
      via Copley (Express)
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it splits the headsign name when it contains 'via'  1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large notranslate"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via notranslate"
    >
      via Harvard Square
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time "
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;
